### ISSUE TITLE
bug: line truncation in neovim terminal requires functionality to log to file

### ISSUE THREAD

**Issue by matu3ba** [stevearc/overseer.nvim#202]
### Neovim version (nvim -v)

NVIM v0.10.0-dev-1178+g34a786bc4

### Operating system/version

Linux ASG148 5.15.0-83-generic

### Describe the bug

![truncated_lines_problem2](https://github.com/stevearc/overseer.nvim/assets/13089667/48afbb1f-3f98-40c5-a6b8-63a0d6f5cd55)

### Steps To Reproduce

Define a command to execute and wait for it to finish.
2. :OverseerToggle!
3. `<CR>` with cursor over job
4. Press 4 for open
5. Create a split, ie `C-w s`
6. Close the split
7. See how long lines are truncated and the content appears to be lost.

### Expected Behavior

Only correct solution with terminal emulator semantics: Line wrapping requires shells sending special line wrap escape sequences and the terminal emulator storing it + handling it correctly.

Multiple solutions:
- 1. Non-Solution is to wait for neovim upstream to do the correct solution (we are waiting already a long time)
- 2. Offer a file-only log mode (with or without separation of stdout, stderr) [can programs have different buffering behavior vs shell?]
- 3. Support external shell + terminal emulator combinations [bad to non-existing neovim integration]
- 4. Hack up the shell[s] to pipe to both file + terminal emulator
- 5. Get from the terminal emulator (library) to the full shell->terminal emulator stream including escape sequences and copy it into a buffer. [requires shells to send correct line wrap escape sequences]

From my point of view the sanest option is to do 2 at this moment. 
If terminal emulators would be sane with correct terminal emulator libraries and neovim having a way to attach them or a terminal emulator offering such functionality, I'd go for 5.

### Minimal example file

Not relevant.

### Minimal init.lua

```Lua
Not relevant.
```

### Additional context

Sidenode: Line truncation in neovim also breaks multiline copies, which just work in most terminals, unless multiplexers are involved.
Addition: The behavior is identical without multiplexer.

**Comment by matu3ba** [comment]
I do see 
```
on_output_write_file                                        *on_output_write_file*
    Write task output to a file

    Parameters:
      {*filename} `string` Name of file to write output to
```
The usual approach is to 1. provide prefix path, 2. use task name and 3. append incrementing number for each run.

I think having at least an example strategy would be nice for users to workaround the behavior. I do only see https://github.com/stevearc/overseer.nvim/blob/6fe36fc338fbeaf35b0f801c25f7f231c431a64b/lua/neotest/client/strategies/overseer.lua#L46, but no strategy does use the function meaning it looks to me like it is untested code(https://github.com/search?q=repo%3Astevearc%2Foverseer.nvim%20on_output_write_file&type=code).

**Comment by stevearc** [comment]
I agree the terminal behavior is not ideal, and I don't expect it to improve any time soon. If you want something quick to test, you could try this strategy:
```lua
require("overseer").setup({
  strategy = { "jobstart", use_terminal = false },
})
```

That will display the output in a normal buffer instead of a terminal buffer. You'll lose some of the nice terminal highlighting, but depending on what kinds of commands you're running it might be fine.

In the medium term, I think there may be something I can do to make the reflow behavior better using logfiles. It probably wouldn't involve `on_output_write_file`, as that was made for a very specific purpose (neotest integration).

Do you want output logs for any other reason, or are you just thinking about this as a way to fix the terminal display issues?

**Comment by matu3ba** [comment]
Merci, I'll try out all the things at work tomorrow.
